### PR TITLE
VERSIONING: Add v2 suffix to the module path

### LINKED ISSUE #4359: Make package Go Module compatible
**Is your feature request related to a problem? Please describe.**

The package is not Go Module compatible. The last version that is "go get-able" is `v1.6.1`:

```shell
$ go get github.com/grafana/loki@v2.3.0
go get: github.com/grafana/loki@v2.3.0: invalid version: module contains a go.mod file, so major version must be compatible: should be v0 or v1, not v2
```

**Describe the solution you'd like**

Make the package Go Module compatible.

**Describe alternatives you've considered**

There are no proper alternatives apart from maintaining a fork.

**Additional context**

Screenshot 2021-09-21 at 11 06 48

### PR DESCRIPTION
**Pull request by lukasmalkmus**
**What this PR does / why we need it**:
It is not possible to get this module using `go get` for the `v2.x` tag. This PR adds the `v2` suffix to the modules import path.

**Which issue(s) this PR fixes**:
Fixes #4359

**Special notes for your reviewer**:
The [preferred](https://go.dev/blog/v2-go-modules) approach according to this, is a different one. However, [go-github](https://github.com/google/go-github/blob/master/scrape/apps.go#L20) chooses another approach that seems to be widely accepted.

This PR should not break anyone as `go get` does not work with the `v2.x` tag anyways, right? RIGHT?

**Checklist**
- [x] Documentation added
- [x] Tests updated
- [ ] Add an entry in the `CHANGELOG.md` about the changes.

### CODE REVIEW

**Comment by kavirajk** [comment]
@lukasmalkmus understand your pain :(  We come across this discussion once in a while. There are some effort in the past to do this as well.

Please take a look at this discussion here. https://github.com/grafana/loki/issues/2826#issuecomment-968709560

tl;dr is even if we plan to support module path in Loki, it supposed to be done on next major release say Loki v3 (as recommended by Go wiki. more details on that thread)

**Comment by kavirajk** [comment]
Closing this PR. but feel free to discuss it in the original issue #2826

**Comment by lukasmalkmus** [comment]
Understood. Is there an eta on a next major realize?

From #2826:

> For example, if you are the author of foo, and the latest tag for the foo repository is v2.2.2,  and foo has not yet adopted modules, then the best practice would be to use v3.0.0 for the first release of foo to adopt modules (and hence the first release of foo to contain a go.mod file).

Just wanna point out for people stumbling accross this, that having a `go.mod` file in a `v2.x` tagged project but not having adopted modules (by using the appropriate import path or working with the sub-package approach) is causing confusion for `go get` and resulting in the error pointed out in #4359.
